dartagnan/src/main/java/com/dat3m/dartagnan/program/processing/compilation/VisitorTso.java

```diff
@@ -29,14 +27,14 @@ public List<Event> visitCreate(Create e) {
                 Store store = newStore(e.getAddress(), e.getMemValue(), "");
                 store.addFilters(C11.PTHREAD);
 
-                return eventSequence(
-                                store);
+                return tagList(eventSequence(
+                                store));
```

Comment:
Do you want to keep atomic tags for the pthread events?
I think an mfence (if even necessary) is more natural.

Reply:
Power (which is the closest architecture in the sense that it relies in fences to implement act-rel) uses fences so I will do the same (not 100% they are needed, but I prefer to play it safe than smart)